gio/src/subclass/async_initable.rs

```diff
@@ -0,0 +1,522 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::object::Cast;
+use glib::thread_guard::ThreadGuard;
+use glib::translate::*;
+use glib::Error;
+
+use glib::subclass::prelude::*;
+
+use std::ptr;
+
+use crate::AsyncInitable;
+use crate::AsyncResult;
+use crate::Cancellable;
+
+use std::{future::Future, pin::Pin};
+
+pub trait AsyncInitableImpl: ObjectImpl {
+    #[allow(unused_variables)]
+    fn init_async<F>(
+        &self,
+        initable: &Self::Type,
+        io_priority: glib::Priority,
+        cancellable: Option<&Cancellable>,
+        callback: Option<F>,
+    ) where
+        F: FnOnce(Result<(), glib::Error>) + 'static,
+    {
+        let main_context = glib::MainContext::ref_thread_default();
+        let initable = initable.clone();
+        main_context.spawn_local(async move {
+            let ret = initable.imp().init_future(&initable, io_priority).await;
+            if let Some(callback) = callback {
+                callback(ret);
+            }
+        });
+    }
+    #[allow(unused_variables)]
+    fn init_future(
```

Comment:
Having only the futures variant seems fine to me

Reply:
Removed it and kept only `init_future`